client/src/components/HealthCareProvider/HCProviderLogin.tsx

```diff
@@ -0,0 +1,46 @@
+import React, { FunctionComponent, useContext } from "react";
+import { HealthContext } from "../../providers/HealthProvider";
+import axios from 'axios';
+import Web3 from 'web3';
+import { AuthContext } from "../../providers/AuthProvider";
+import { useNavigate } from "react-router-dom";
+
+const HCProviderLogin: FunctionComponent<{}> = () => {
+    const { currentAccount } = useContext(HealthContext);
+    const { login } = useContext(AuthContext);
+    const navigate = useNavigate();
+    const loginhandler = async () => {
+        console.log('login')
+        let resp = await axios.get('/custom_auth/' + currentAccount);
+
+        if (resp && resp.data) {
+            const provider = new Web3.providers.HttpProvider('http://127.0.0.1:8545');
+            const web3 = new Web3(provider);
+            const account = currentAccount || "";
+
+            const signed = web3.utils.keccak256(resp.data.challenge);
+            console.log(signed)
+            if (signed) {
+                web3.eth.sign(signed, account, async (err, signature) => {
+                    let auth_resp = await axios.get('/verify_auth/' + signature + '?client_address=' + account);
```

Comment:
this calls the patient login endpoint, you can create a new endpoint for provider and use that.